php/libraries/FilesUploadHandler.class.inc

```diff
@@ -0,0 +1,226 @@
+<?php declare(strict_types=1);
+
+/**
+ * Class to upload files in a safe way.
+ *
+ * This class handles files being uploaded to LORIS. It should
+ * serve as the ONLY way that files are uploaded so that all file uploading
+ * functionality can occur on a well-tested foundation.
+ *
+ * PHP Version 7
+ *
+ *  @category Main
+ *  @package  Behavioural
+ *  @author   John Saigle <john.saigle@mcin.ca>
+ *  @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ *  @link     https://www.github.com/aces/Loris
+ */
+namespace LORIS;
+
+use \Psr\Http\Message\ResponseInterface;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * Class to validate file uploads to LORIS.
+ *
+ *  @category Main
+ *  @package  Behavioural
+ *  @author   John Saigle <john.saigle@mcin.ca>
+ *  @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ *  @link     https://www.github.com/aces/Loris
+ */
+class FilesUploadHandler implements RequestHandlerInterface
+{
+
+    /**
+     * The target upload directory.
+     *
+     * @var \SplFileInfo
+     */
+    protected $uploadDirectory;
+
+    /**
+     * A whitelist of all MIME types to accept.
+     *
+     * @var string[]
+     */
+    protected $permittedMIMETypes = array();
+
+    /**
+     * A UNIX permissions octet.
+     *
+     * @var int
+     */
+    protected $permissions = 0644;
+
+    /**
+     * Whether existing files should be overwritten.
+     *
+     * @var bool
+     */
+    protected $overwrite = false;
+
+    /**
+     * Create new instance of a File Uploader.
+     *
+     * @param \SplFileInfo $uploadDirectory The target upload directory
+     */
+    public function __construct(\SplFileInfo $uploadDirectory)
```

Comment:
Just a comment/suggestion
The constructor signature should/could be 
```
public function __construct(\SplFileInfo $uploadDirectory, array $permittedMIMETypes = [], int $permissions = 0644, bool $overwrite = false )
``` 
That way the default values for the class variables could be removed but you would need to set them in the constructor code block. I think it facilitates the instantiation for non-default values.
Feel free to change it or not (It's your PR.)